spec/04-basic-declarations-and-definitions.md

```diff
@@ -840,6 +840,23 @@ class C extends I {
 Here, it is OK to leave out the result type of `factorial`
 in `C`, even though the method is recursive.
 
+### Tail-call Elimination
+
+Method definitions which contain self-recursive invocations in tail position
+are optimized for stack safety. Self-invocations which are the last operation
+before returning from the method are replaced with jumps to the beginning of
+the method, much as in a while loop.
+
+This transform is performed automatically, without intervention by the user,
+but may be enforced using `scala.annotation.tailrec`, for cases where
```

Comment:
Slightly ambiguous? "may be enforced" could mean "the user can enforce" or "the compiler may enforce".